src/bgg_companion_api.py

```diff
@@ -42,22 +44,27 @@ def __to_board_game(item: collections.OrderedDict) -> BoardGame:
     description = item["description"]
     minplayers = item["minplayers"]["@value"]
     maxplayers = item["maxplayers"]["@value"]
-    thumbnail = item["thumbnail"]
-    image = item["image"]
+    if "thumbnail" in item and "image" in item:
+        thumbnail = item["thumbnail"]
+        image = item["image"]
+    else:
+        thumbnail = None
+        image = None
     if isinstance(item["name"], OrderedDict):
         name = item["name"]["@value"]
     elif isinstance(item["name"], list):
         for a_dict in item["name"]:
             if a_dict["@type"] == "primary":
                 name = a_dict["@value"]
+                break
     else:
         name = None
     return BoardGame(id=id, name=name, description=description, type=type, minplayers=minplayers, maxplayers=maxplayers,
                      thumbnail=thumbnail, image=image)
 
 
-def get_object_ids(user: str) -> list[str]:
-    users_game_collection = get_users_collection(user)
+def get_users_game_ids(user: str) -> list[str]:
+    users_game_collection = get_collection(user)
```

Comment:
Now that `get_collection` can raise a `BoardGameIsNoneError` error here: https://github.com/JDGiardino/BGG-Companion/pull/16/files#diff-835bd37d2f749fae74d5c53432fdf3fe53e09b79bbdb50dd64c0148f399a5591R23 do you think it would be worth catching that exception here? Or do you want the caller of `get_users_game_ids` to handle the `BoardGameIsNoneError` error themselves?

Reply:
Hmmmmm yeah this really has me thinking now.  I guess the high level question is... who should be responsible for raising errors, the caller of a function with the error or the function itself?  I don't know what best practice is but I think it's a matter of separation of duties and single responsibility for functions.  In this example : 

On one hand the get_collection function has 1 job and that is to make a call for a collection from BGG api based on a passed user.  So maybe it shouldn't have to worry about the contents of the response, just return whatever it gets. 

On the other hand though, why should it be up to the callers to care about this?  In this example at least, the caller would have to know some very specific details of the API data shape that maybe it shouldn't have to and that knowledge and responsibility should be contained within the get_collection function.  It's responsible for making an API call AND understanding some details of that API that callers shouldn't have to.  Additionally if there were multiple callers of this function that's multiple places to be checking for this error rather than just within the function itself. 

I'm not really sure what is best, but based on what I've thought of us far it feels like this responsibility should remain in get_collection.  What do you think @nickdibari ?

Reply:
And if that logic is sound, I wonder if https://github.com/JDGiardino/BGG-Companion/pull/16/files#diff-835bd37d2f749fae74d5c53432fdf3fe53e09b79bbdb50dd64c0148f399a5591R71 makes sense to be handled there instead of inside the get_collection function.

Reply:
I think it's totally up to you, I've seen designs go both ways and there are pros/cons to both. Personally I lean towards the second point you made: a caller should be able to call a function and immediately be able to use the return value for its own purpose without having to worry about checking the return value for exceptional data or other things. This is similar to the [Design by contract](https://en.wikipedia.org/wiki/Design_by_contract) pattern where functions and methods have _contracts_ that callers expect them to conform to, with predictable behavior for a given input/context.

I think this might be something we can hash out in person, as it's more of a philosophical discussion than a particular comment to this PR. Hopefully we can come to a consensus for you and then you'd not only be able to resolve this but also have a better sense for what code designs work best for you!

Reply:
Again, this is another topic I'm really glad we talked through.  And Design by Contract totally makes sense and sounds like a best practice I would like to follow.  We'll definitely continue this conversation in our next in person chat.